rust-toolchain

```diff
@@ -1,5 +0,0 @@
-[toolchain]
-channel = "nightly-2022-05-05"
-components = [ "rustfmt", "clippy" ]
-targets = [ "wasm32-unknown-unknown" ]
-profile = "minimal"
```

Comment:
Do we need this anymore? I noticed other projects have not included this for quite a while now.

If we need it, maybe we should add some brief comments in it about what purpose it serves.

Reply:
The CI uses rustup show to install the required rust toolchain. Maybe it can be changed to use `rustup default $toolchain` or `rustup override set $toolchain`

The CI fails now because the default toolchain does not match what is required, but the required one is installed on there so just need to switch to it.
```
Run rustup show
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/gh-actions/.rustup

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu (default)
nightly-2022-02-20-x86_64-unknown-linux-gnu
nightly-2022-04-20-x86_64-unknown-linux-gnu
nightly-2022-05-05-x86_64-unknown-linux-gnu

active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.63.0 (4b91a6ea[7](https://github.com/PureStake/moonbeam/actions/runs/3101551623/jobs/5023033279#step:4:8) 2022-0[8](https://github.com/PureStake/moonbeam/actions/runs/3101551623/jobs/5023033279#step:4:9)-08)
```